### PR TITLE
Add full stack example with Flutter frontend and Cloud Run backend

### PR DESCRIPTION
**Pull request by subfuzion**
Very simple demo that shows a Flutter app using a Dart functions app for a greeting API. The README docs detail building and testing locally and then deploying to Cloud Run.

This pull requests also includes community contributions that were made in [here](https://github.com/subfuzion/dart-flutter-fullstack-demo/pulls?q=is%3Apr+is%3Aclosed). Thanks!

### CODE REVIEW

**Comment by kevmoo** [comment]
What can we do to get CI validating this?

File: examples/fullstack/frontend/lib/config.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+import 'dart:convert';
+
+import 'package:flutter/services.dart';
+
+enum Environment {
+  dev,
+  prod,
+}
+
+extension enumToString on Environment {
```
Comment (kevmoo) [inline examples/fullstack/frontend/lib/config.dart:24]: Name this like a type `EnumToString`

File: examples/fullstack/frontend/lib/config.dart [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+  String toName() {
+    return this.toString().split('.').last;
+  }
+}
+
+extension stringToEnvironment on String {
```
Comment (kevmoo) [inline examples/fullstack/frontend/lib/config.dart:30]: I know extension types are nice, but this not ideal. We're adding this function to EVERY String, when it's only really useful is some targeted places.

File: examples/fullstack/frontend/lib/services/api.dart [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      headers: {'content-type': 'application/json'},
+      body: body,
+    );
+
+    if (res.statusCode == HttpStatus.ok) {
+      var greetingResponse = GreetingResponse.fromJson(jsonDecode(res.body));
+      return greetingResponse;
+    }
+
+    throw Future.error(Exception(
```
Comment (kevmoo) [inline examples/fullstack/frontend/lib/services/api.dart:46]: Just throw here. No need to wrap it in a `Future.error`

File: examples/fullstack/frontend/lib/services/api_types.dart [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  final String name;
+
+  GreetingResponse({required this.salutation, required this.name});
+
+  factory GreetingResponse.fromJson(Map<String, dynamic> json) =>
+      _$GreetingResponseFromJson(json);
+
+  Map<String, dynamic> toJson() => _$GreetingResponseToJson(this);
+
+  @override
```
Comment (kevmoo) [inline examples/fullstack/frontend/lib/services/api_types.dart:50]: are these ever compared or put in a hashtable?

**Review by kevmoo — APPROVED** [review]

File: examples/fullstack/frontend/lib/services/api_types.dart (hunk 4 above)
Comment (subfuzion) [reply examples/fullstack/frontend/lib/services/api_types.dart:50]: These are artifacts from the basic JSON example where comparisons are made during testing. However, that's not the case here, so I can remove.

File: examples/fullstack/README.md [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+```
+
+The following assumes running on the macOS desktop. See Flutter docs for
+building for Windows or Linux desktops.
+
+If you have not enabled desktop support for your Flutter installation,
+do so now:
+
+```shell
+flutter config --enable-macos-desktop
```
Comment (domesticmouse) [inline examples/fullstack/README.md:111]: Please add all desktops here:

```
flutter config --enable-windows-desktop
flutter config --enable-macos-desktop
flutter config --enable-linux-desktop
```

File: examples/fullstack/README.md [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+```text
+Setting "enable-macos-desktop" value to "true".
+
+You may need to restart any open editors for them to read new settings.
+```
+
+### Connect frontend to backend running on local machine
+
+```shell
+flutter run -d macos
```
Comment (domesticmouse) [inline examples/fullstack/README.md:125]:
```
flutter run -d windows
flutter run -d macos
flutter run -d linux
```

File: examples/fullstack/README.md [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+
+First update your `prod` configuration with your Cloud Run project's
+ServiceURL in `frontend/assets/config/prod.json` (note that this file
+is in `frontend/.gitignore`; if you want to commit your change you
+will need to remove or comment out the entry).
+
+Then when running `flutter run` define our environment as `prod`:
+
+```shell
+flutter run -d macos --dart-define=ENV=prod
```
Comment (domesticmouse) [inline examples/fullstack/README.md:141]: ditto

File: examples/fullstack/frontend/.metadata [hunk 8]
```diff
@@ -0,0 +1,10 @@
+# This file tracks properties of this Flutter project.
+# Used by Flutter tool to assess capabilities and perform upgrades etc.
+#
+# This file should be version controlled and should not be manually edited.
+
+version:
+  revision: 8264cb3e8a797eef39cbcd32bb56fd07790efb7f
+  channel: dev
```
Comment (domesticmouse) [inline examples/fullstack/frontend/.metadata:8]: Is there anything dev specific in this app, or should it work with flutter stable channel?

**Review by domesticmouse — APPROVED** [review]

File: examples/fullstack/frontend/lib/config.dart (hunk 2 above)
Comment (subfuzion) [reply examples/fullstack/frontend/lib/config.dart]: Just a nit, but really every string has *access* to the function. Wouldn't do this for a library, just a utility for the app, but not to be obtuse, I agree we should be showing best practices. Thanks.

File: examples/fullstack/README.md (hunk 5 above)
Comment (subfuzion) [reply examples/fullstack/README.md:111]: Thought you were gonna give me a PR! I see how it is... :D

File: examples/fullstack/frontend/.metadata (hunk 8 above)
Comment (subfuzion) [reply examples/fullstack/frontend/.metadata]: Yes ... `clearSnackBars` requires dev channel. I'd like to leave it, but commented it out for now.

**Review by kevmoo — APPROVED** [review]

**Pull request merged**